roles/servicetelemetry/tasks/component_qdr.yml

```diff
@@ -227,7 +227,7 @@
             prefix: ceilometer
         edgeListeners:
           - expose: true
-            host: 0.0.0.0
+            host: ''
```

Comment:
Case 1.  contradicts the output above where we're shown it listening to both IPv4 and IPv6. I'm wondering how you did your test - was it via this change in the operator, or did you configure the QDR directly? My gut feeling is that maybe you put `''` literally in the QDR config file directly, and the operator processes `''` as a YAML literal string, not passing them through to the actual config file. Just a guess.

I can't explain Case 2, it looks right from here.

If I get a chance today I'll try to test this change as well.

Reply:
This test is not official since my local build environment was giving me trouble today, but:

* I edited my Interconnect object to say `host: ''` like in this patch
* OCP returns it to me with empty double-quotes instead
```
$ oc get Interconnect -o yaml | grep host
      host: ""
```
* The operator renders the config file with no host field at all
 ```
$ oc rsh pod/default-interconnect-574d468757-wshtn grep -C 4 "port: 5671" /opt/interconnect/etc/qdrouterd.conf
}

listener {
    role: edge
    port: 5671
    saslMechanisms: PLAIN
    authenticatePeer: true
    sslProfile: openstack
}

```
* qdr is  listening on both protocols
```
$ oc rsh pod/default-interconnect-574d468757-wshtn netstat -anp | grep :5671
tcp        0      0 0.0.0.0:5671            0.0.0.0:*               LISTEN      1/qdrouterd         
tcp6       0      0 :::5671                 :::*                    LISTEN      1/qdrouterd 
```

Note that this was done on top of upstream builds and by manually editing the Interconnect object, not from actually running this PR code. I'll need to figure out what's wrong with my cluster; I think I'm just using it for too many things at once.

Reply:
I also tried this with no host field in the Interconnect object at all, and the results were the same. Maybe that's the preferred way to write it to prevent any future doubts about the problems Martin has shown.